src/java/main/org/apache/zookeeper/server/quorum/Leader.java

```diff
@@ -1376,4 +1382,8 @@ public static String getPacketType(int packetType) {
             return "UNKNOWN";
         }
     }
+
+    private boolean isRunning() {
+        return self.isRunning() && zk.isRunning();
```

Comment:
Ok, so this is currently only used in line 588, so the correct behavior is to shut down in the case either is not running, yes?